src/main/java/core/basesyntax/service/RegistrationServiceImp.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+
+public class RegistrationServiceImp {
+    private final StorageDao storageDao;
+
+    public RegistrationServiceImp(StorageDaoImpl storage) {
+        this.storageDao = storage;
+    }
+
+    public void register(User user) {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserDataException(
+                    "User with this login already exists");
+        }
+        if (user.getLogin().length() < 6) {
+            throw new InvalidUserDataException(
+                    "Login must be at least 6 characters long");
+        }
+        if (user.getPassword().length() < 6) {
```

Comment:
Similar to the user's password, you need to check if it's null before calling the length method on it. If it's null, it will throw a NullPointerException. Instead, you should throw an InvalidUserDataException with an informative message.